### PR TITLE
Use fallocate to allocate the images when creating.

### PR DESCRIPTION
**Pull request by wpoely86**
Changes proposed in this pull request

 -  Switch to using `posix_fallocate` for allocation of image files
 -  Handle EINTR
 -  Fix warnings by Clang

@singularityware-admin

### CODE REVIEW

File: src/lib/image/bind/bind.c [hunk 1]
```diff
@@ -99,10 +99,7 @@ int _singularity_image_bind(struct image_object *image) {
 #endif
 
     singularity_message(DEBUG, "Calculating image offset\n");
-    if ( ( lo64.lo_offset = singularity_image_offset(image) ) < 0 ) {
-        singularity_message(ERROR, "Could not obtain message offset of image\n");
-        ABORT(255);
-    }
+    lo64.lo_offset = singularity_image_offset(image);
```
Comment (wpoely86) [inline src/lib/image/bind/bind.c:90]: `lo_offset` is unsigned and `singularity_image_offset` never returns anything less then 0.

Comment (gmkurtzer) [reply src/lib/image/bind/bind.c:90]: fair enough. I left that in there though incase we update singularity_image_offset() to return a negative number on error. But in that case, we can look for calls to it and update accordingly.

File: src/lib/image/create/create.c [hunk 2]
```diff
@@ -57,20 +53,42 @@ int _singularity_image_create(struct image_object *image, long int size) {
...
     singularity_message(VERBOSE2, "Growing image to %ldMB\n", size);
-    // TODO: there are likely better ways to do this (falloc?); further, we should really handle
-    // EINTR here.
-    for(i = 0; i < size; i++ ) {
-        if ( fwrite(buff, 1, BUFFER_SIZE, image_fp) < BUFFER_SIZE ) {
-            singularity_message(ERROR, "Failed allocating space to image: %s\n", strerror(errno));
-            ABORT(255);
+    while(1)
+    {
+        retval = posix_fallocate(image->fd, sizeof(LAUNCH_STRING), size*BUFFER_SIZE);
```
Comment (gmkurtzer) [inline src/lib/image/create/create.c]: I forget (and the man page didn't seem very clear on this), does posix_fallocate() zero out the allocated disk space, or is it sparse? Thanks!

Comment (wpoely86) [reply src/lib/image/create/create.c]: As far as I understand, it's filled with zeros. A quick test seems to confirm this.

**Comment by wpoely86** [comment]
As reference: http://www.lucas-nussbaum.net/blog/?p=331#comment-21515

So, `posix_fallocate` should not create a sparse file.

**Comment by gmkurtzer** [comment]
This has been merged into the development branch. Thank you!

**Pull request closed without merging**